web/src/plugins/logs/SearchBar.vue

```diff
@@ -1205,31 +1205,27 @@ export default defineComponent({
       getSuggestions();
     };
 
-    function getColumnNames(data) {
+    const getColumnNames = (columnData: any) => {
       const columnNames = [];
-      data.forEach((item) => {
-        if (item.expr && item.expr.column) {
-          columnNames.push(item.expr.column);
-        } else if (item.expr && item.expr.args && item.expr.args.expr) {
-          if (item.expr.args.expr.column) {
-            columnNames.push(item.expr.args.expr.column);
+      for (const item of columnData) {
+        if (item.expr.type === "column_ref") {
+          columnNames.push(item.expr.column.expr.value);
+        } else if (item.expr.type === "aggr_func") {
+          if (item.expr.args.expr.hasOwnProperty("column")) {
+            columnNames.push(item.expr.args.expr.column.value);
           } else if (item.expr.args.expr.value) {
             columnNames.push(item.expr.args.expr.value);
           }
-        } else if (
-          item.expr &&
-          item.expr.name &&
-          item.expr.type === "function"
-        ) {
+        } else if (item.expr.type === "function") {
           item.expr.args.value.map((val) => {
             if (val.type === "column_ref") {
-              columnNames.push(val.column);
+              columnNames.push(val.column?.expr?.value);
             }
           });
         }
-      });
+      }
       return columnNames;
-    }
+    };
```

Comment:
ColumnData type will be dynamic for different function. For e.g histogram function will have two arguments but another custom function might have different number of arguments and accordingly it will provide nested level of JSON object.